submissions/VBystrov/friends-app/js/index.js

```diff
@@ -0,0 +1,328 @@
+
+    this.friendlist.applySearch(this.parameters);
+    this.friendlist.showCards();
+  }
+
+  changeSearchAge({ target }) {
+    this.parameters.age = parseInt(target.value, 10);
+
+    this.setUrl('age', target.value);
+
+    this.friendlist.applySearch(this.parameters);
+    this.friendlist.showCards();
+  }
+
+  changeSearchGender() {
+    const selectedGender = this.genderContainer.querySelector(
+      '.search-gender__input:checked'
+    ).value;
+    if (selectedGender !== this.parameters.gender) {
+      this.setUrl('gender', selectedGender);
+      this.parameters.gender = selectedGender;
+
+      this.friendlist.applySearch(this.parameters);
+      this.friendlist.showCards();
+    }
+  }
+
+  changeSorting({ target }) {
+    const sortingControl = target.closest('.search__sort');
+    if (sortingControl) {
+      this.parameters.sorting = sortingControl.value;
+
+      this.setUrl('sorting', sortingControl.value);
+
+      this.friendlist.applySorting(this.parameters.sorting);
+      this.friendlist.showCards();
+
+      this.sortingControlActive.classList.remove('search__sort_active');
+      this.sortingControlActive = sortingControl;
+      this.sortingControlActive.classList.add('search__sort_active');
+    }
+  }
+}
+
+class Card {
+  constructor(cardData) {
+    this.firstName = cardData.name.first;
+    this.lastName = cardData.name.last;
+    this.email = cardData.email;
+    this.picture = cardData.picture.large;
+    this.country = cardData.location.country;
+    this.city = cardData.location.city;
+    this.age = cardData.dob.age;
+    this.gender = cardData.gender;
+    this.generateHtml();
+  }
+
+  generateHtml() {
+    const iconMale = 'fa-mars';
+    const iconFemale = 'fa-venus';
+    const iconOther = 'fa-genderless';
+    let iconType;
+    switch (this.gender) {
+      case 'male':
+        iconType = iconMale;
+        break;
+      case 'female':
+        iconType = iconFemale;
+        break;
+      default:
+        iconType = iconOther;
+        break;
+    }
+
+    this.html = `<div class="friendcard" title="${this.firstName} ${this.lastName}">
+    <header class="friendcard-header">
+      <h3 class="friendcard-header__name">${this.firstName} ${this.lastName}</h3>
+      <a title="${this.email}" href="mailto:${this.email}">
+        <i
+          class="far fa-envelope fa-lg friendcard-header__email-icon"
+        ></i>
+      </a>
+    </header>
+    <img
+      class="friendcard__avatar"
+      src="${this.picture}"
+      alt="human face"
+    />
+    <div class="location-wrapper">
+      <div class="friendcard__location">
+        <p>${this.country}</p>
+        <p>${this.city}</p>
+      </div>
+      <div class="gender-age-wrapper">
+        <i class="fas ${iconType} fa-lg gender-icon"></i>
+        <p class="friendcard__age">${this.age}</p>
+      </div>
+    </div>
+  </div>`;
+  }
+}
+
+class Sorters {
+  constructor() {
+    this.nameAZ = function (card1, card2) {
```

Comment:
Don’t use names as arr, array, data or etc. They to generic. Don’t use things like sortingByABC and sortingByZYX - it’s unclear and just weird. user1 and user2 - don't use numbers in names. Just don't :) friendsArr also a bad idea - you show data type in the variable name and in JS it’s redundant